### PR TITLE
feat: 864dmtze8 - apy calculation on frontend [FE]

### PR DESCRIPTION
**Pull request by naruto0913**
Main updates
1. add apy calculation files in utils/apy
2. fix minor bugs in frontend code

### CODE REVIEW

File: ui/hedgepie.finance/utils/apy/biswap.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import Moralis  from 'moralis';
+import { EvmChain } from '@moralisweb3/common-evm-utils';
+
+import { BLOCK_GENERATION_TIME, SECONDS_PERY_YEAR } from 'constants/common'
+
+import lpAbi from 'config/abi/LPPair.json'
+import masterChefAbi from 'config/abi/BiswapMasterChef.json'
+
+const PRICE_URL = "https://api.binance.com/api/v3/avgPrice?symbol="
+const RPCURL = "https://bsc-dataseed1.binance.org"
```
Comment (daiki-iu) [inline ui/hedgepie.finance/utils/apy/biswap.ts:12]: @naruto0913  this should be PRC_URL.

File: ui/hedgepie.finance/utils/apy/biswap.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+const MASTERCHEF = "0xDbc1A13490deeF9c3C12b44FE77b503c1B061739"
```
Comment (daiki-iu) [inline ui/hedgepie.finance/utils/apy/biswap.ts:13]: @naruto0913  this should be masterChefAddr.

File: ui/hedgepie.finance/utils/apy/biswap.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+const rpcProvider = new ethers.providers.JsonRpcProvider(RPCURL);
+const masterchefContract = new ethers.Contract(
+    MASTERCHEF,
+    masterChefAbi,
+    rpcProvider
+);
+
+const startMoralis = async () => {
+    await Moralis.start({
+        apiKey: 'zUkiTWSfy0i9TEaViuSErFkpStIojZMuhWxBUMDp1GrYU8HSKutYwdYFYZp7y291'
```
Comment (daiki-iu) [inline ui/hedgepie.finance/utils/apy/biswap.ts:24]: @naruto0913  apiKey value should be moved to .env. Let's add a env var both in .env.example and .env as MORALIS_API_KEY.

File: ui/hedgepie.finance/utils/apy/biswap.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        );
+        
+        const [lpTokenDeposited, lpTokenTotalSupply, reserve0, token0] = await Promise.all([
+            lpTokenContract.balanceOf(MASTERCHEF),
+            lpTokenContract.totalSupply(),
+            lpTokenContract.getReserves(),
+            lpTokenContract.token0()
+        ]);
+
+        const BSWPrice = await axios.get(`${PRICE_URL}BSWUSDT`);
```
Comment (daiki-iu) [inline ui/hedgepie.finance/utils/apy/biswap.ts:59]: @naruto0913  let's change var name to bswPrice.

File: ui/hedgepie.finance/utils/apy/index.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    let apyVal: any
+    if (adapter.includes('apeswap')) {
+        apyVal = await getApeswapAPY(params.chain ? params.chain : 56, params.pid ? params.pid : 3)
+        // console.log('apeswap', apyVal, "eeeeeeeeeeeeeeeeee")
+    } else if (adapter.includes("biswap")) {
+        apyVal = await getBiswapAPY(params.pid ? params.pid : 9)
+        // console.log('biswap', apyVal, "eeeeeeeeeeeeeeeeee")
+    } else if (adapter.includes("autofarm")) {
+        apyVal = await getAutofarmAPY(params.chain ? params.chain : 'bsc', params.pid ? params.pid : 620)
+        // console.log('autofarm', apyVal, "eeeeeeeeeeeeeeeeee")
```
Comment (daiki-iu) [inline ui/hedgepie.finance/utils/apy/index.ts:15]: @naruto0913  let's remove commented console logs here.

File: ui/hedgepie.finance/v2/components/Leaderboard/LeaderboardContent.tsx [hunk 6]
```diff
@@ -47,6 +51,10 @@ function LeaderboardContent() {
         })
 
       let tokens = [] as TokenInfo[]
+      const bnbPrice = await getPrice('BNB')
+
```
Comment (daiki-iu) [inline ui/hedgepie.finance/v2/components/Leaderboard/LeaderboardContent.tsx:55]: @naruto0913  no empty line is needed here.

**Review by daiki-iu — CHANGES_REQUESTED** [review]
@naruto0913  LGTM, but left some comments.

**Review by sdscode00 — APPROVED** [review]
LGTM 👍

**Review by daiki-iu — APPROVED** [review]

**Review by daiki-iu — APPROVED** [review]

**Pull request merged**